### PR TITLE
Start of network protocol, chunk loading and bevy 0.13 update

### PR DESCRIPTION
**Pull request by eigenraven**
A bunch of unfinished features here, but I'd rather not keep rebasing this over and over.

### CODE REVIEW

File: crates/ocg_common/src/network/server.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    ) -> Promise<(), Error> {
+        let title = "OCG Server";
+        let subtitle = "Subtitles to be implemented!";
+        let mut meta = results.get().init_metadata();
+        let mut ver = meta.reborrow().init_server_version();
+        ver.set_major(0);
+        ver.set_minor(0);
+        ver.set_patch(1);
+        ver.set_build("todo");
+        ver.set_prerelease("");
```
Comment (screret) [inline crates/ocg_common/src/network/server.rs:62]: maybe these should be `const`s somewhere?

File: crates/ocg_common/src/network/server.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        meta.set_title(title);
+        meta.set_subtitle(subtitle);
+        meta.set_player_count(0);
+        meta.set_player_limit(12);
```
Comment (screret) [inline crates/ocg_common/src/network/server.rs:67]: same for these

File: crates/ocg_common/src/voxel/biomes.rs [hunk 3]
```diff
@@ -48,7 +57,9 @@ pub fn setup_basic_biomes(biome_registry: &mut BiomeRegistry) {
                 let (i_grass, _) = block_registry.lookup_name_to_object(GRASS_BLOCK_NAME.as_ref()).unwrap();
                 let (i_dirt, _) = block_registry.lookup_name_to_object(DIRT_BLOCK_NAME.as_ref()).unwrap();
                 let (i_stone, _) = block_registry.lookup_name_to_object(STONE_BLOCK_NAME.as_ref()).unwrap();
-                let (i_snow_grass, _) = block_registry.lookup_name_to_object(SNOWY_GRASS_BLOCK_NAME.as_ref()).unwrap();
+                let (i_snow_grass, _) = block_registry
+                    .lookup_name_to_object(SNOWY_GRASS_BLOCK_NAME.as_ref())
+                    .unwrap();
```
Comment (screret) [inline crates/ocg_common/src/voxel/biomes.rs:62]: why different?

File: crates/ocg_common/src/voxel/generator/mod.rs [hunk 4]
```diff
@@ -312,23 +373,28 @@ impl StdGenerator {
...
         let triangles = &voronoi.triangulation().triangles;
-        let triangles: Vec<[&Point; 3]> = (0..triangles.len() / 3).map(|t| [
-            &points[triangles[3 * t + 0]],
-            &points[triangles[3 * t + 1]],
-            &points[triangles[3 * t + 2]]]
-        ).collect();
+        let triangles: Vec<[&Point; 3]> = (0..triangles.len() / 3)
+            .map(|t| {
+                [
+                    &points[triangles[3 * t]],
```
Comment (screret) [inline crates/ocg_common/src/voxel/generator/mod.rs:385]: I thought the `t + 0` looked cleaner.

**Review by screret — CHANGES_REQUESTED** [review]

File: crates/ocg_common/src/network/server.rs (hunk 2 above)
Comment (eigenraven) [reply crates/ocg_common/src/network/server.rs:67]: These will become user-configurable once I add a gui, they shouldn't really be constants

Comment (screret) [reply crates/ocg_common/src/network/server.rs:67]: I see.

File: crates/ocg_common/src/voxel/biomes.rs (hunk 3 above)
Comment (eigenraven) [reply crates/ocg_common/src/voxel/biomes.rs:62]: It overflowed the line length limit, `cargo fmt` updated the formatting everywhere. It's at 120 and I'd rather not make it longer

File: crates/ocg_common/src/network/server.rs (hunk 1 above)
Comment (eigenraven) [reply crates/ocg_common/src/network/server.rs]: Done

File: lib/ocg_schemas/src/voxel/generation/fbm_noise.rs
[the remark this answers is not in the document]
Comment (eigenraven) [reply lib/ocg_schemas/src/voxel/generation/fbm_noise.rs:1]: It breaks references in mod.rs, I'll leave moving this out to your future PRs

File: crates/ocg_common/src/voxel/generator/mod.rs (hunk 4 above)
Comment (eigenraven) [reply crates/ocg_common/src/voxel/generator/mod.rs:385]: It does look cleaner, but then I either have to add a linter suppression rule for this line, and I'd rather not disable warning about useless arithmetic operations globally as that catches some bugs.

File: lib/ocg_schemas/src/voxel/generation/fbm_noise.rs
[the remark this answers is not in the document]
Comment (screret) [reply lib/ocg_schemas/src/voxel/generation/fbm_noise.rs:1]: oh right that's why I didn't do it

File: crates/ocg_common/src/voxel/generator/mod.rs (hunk 4 above)
Comment (screret) [reply crates/ocg_common/src/voxel/generator/mod.rs:385]: 👍

**Review by screret — APPROVED** [review]

**Pull request merged**